File: Hash Table/599. Minimum Index Sum of Two Lists/min_index_basic.cpp
class Solution
{
public:
    vector<string> findRestaurant(vector<string> &list1, vector<string> &list2)
    {
        vector<string> ans;
        unordered_map<string, int> hashmap;

        for (int i = 0; i < list1.size(); i++)
        {
            for (int j = 0; j < list2.size(); j++)
            {
                if (list2[j] == list1[i])
                {
                    hashmap[list2[j]] = i + j;
                }
            }
        }

        int min_val = INT_MAX;
        for (auto item : hashmap)
        {
            if (item.second <= min_val)
            {
                min_val = item.second;
            }
        }

        for (auto item : hashmap)
        {
            if (item.second == min_val)
            {
                ans.push_back(item.first);
            }
        }
        return ans;
    }
};

File: Hash Table/599. Minimum Index Sum of Two Lists/min_index_basic.py
class Solution:
    def findRestaurant(self, list1: List[str], list2: List[str]) -> List[str]:
        ans = []
        hashmap = {}

        for i in range(len(list1)):
            for j in range(len(list2)):
                if list2[j] == list1[i]:
                    hashmap[list2[j]] = i + j

        min_val = 2000
        for key, value in hashmap.items():
            print(key, value)
            if value <= min_val:
                min_val = value

        print(min_val)

        for key, value in hashmap.items():
            if value == min_val:
                print(key, value)
                ans.append(key)

        return ans